tests/testutils/goroutineassistant.go

```diff
@@ -46,14 +50,33 @@ func (a *GoroutineAssistant) Done() {
 }
 
 func (a *GoroutineAssistant) Wait() {
-	done := make(chan struct{})
 	go func() {
 		a.wg.Wait()
-		done <- struct{}{}
+		a.s <- nil
 	}()
-	select {
-	case s := <-a.s:
-		a.t.Fatalf(s)
-	case <-done:
+	err := <-a.s
+	if err == nil {
+		// success
+		return
+	}
+	// Let's mimic testing.Fatalf()'s behavior and fatal with the first received error.
+	// All other goroutines will be killed.
```

Comment:
Yes, because the test related goroutines are spawned by the test's main goroutine. `t.Fatal()` attempts to tear them down immediately. at the end of the main gouroutine the deferred call to `ctx.cleanup()` is executed.